src/esp/gfx/replay/Player.cpp

```diff
@@ -188,16 +198,18 @@ void Player::applyKeyframe(const Keyframe& keyframe) {
     createdInstances_[instanceKey] = node;
   }
 
-  for (const auto& deletionInstanceKey : keyframe.deletions) {
-    const auto& it = createdInstances_.find(deletionInstanceKey);
-    if (it == createdInstances_.end()) {
-      // missing instance for this key, probably due to a failed instance
-      // creation
-      continue;
-    }
+  if (!frameCleared) {
```

Comment:
It might be more readable to clear `deletions` above and then leave this part of the function untouched.

Reply:
The keyframe is a const ref, and should remain so because of the function usage. Given this, I think that it is clearer to use a condition.